api/src/services/items.ts

```diff
@@ -49,6 +54,21 @@ export class ItemsService<Item extends AnyItem = AnyItem> implements AbstractSer
 		return this;
 	}
 
+	createMutationTracker(initialCount = 0): MutationTracker {
+		let mutationCount = initialCount;
+		return {
+			trackMutations(count: number) {
+				mutationCount += count;
+				if (mutationCount > env['MAX_BATCH_MUTATION']) {
```

Comment:
Let's explicitly cast this to a number to prevent weird comparisons if it happens to be a string. Perhaps we should move the env parsing/sanity checking in the constructor and use the parsed value here.

When setting an invalid number this gets parsed to `NaN` and then this comparison is always false.